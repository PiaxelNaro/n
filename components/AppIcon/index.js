// @flow
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import FlexGrid from '../FlexGrid';

/**
 * <AppIcon/>
 * @param {String} src
 * @param {String} size big | small | medium
 * @param {String} store
 * @param {Boolean} showStoreIcon show store icon at the bottom right
 * @example  <AppIcon size={size} store={store} showStoreIcon={Boolean(true)} />
 */
// There is no standard app icon size,
// so we could not difine the specific width and height of icon <img />
const Icon = styled.img`
    width: 100%;
    height: auto;
`;
Icon.Wrapper = styled(FlexGrid)`
    position: relative;
    overflow: hidden;
    &.small {
        width: 22px;
        height: 22px;
    }
    &.medium {
        width: 34px;
        height: 34px;
    }
    &.big {
        width: 64px;
        height: 64px;
    }
    &.ios > img {
        border-radius: 15.625%;
    }
    &.appletv.small > img {
        border-radius: 1px;
    }
    &.appletv.medium > img {
        border-radius: 2px;
    }
    &.appletv.big > img {
        border-radius: 4px;
    }
`;
Icon.StoreOverlay = styled.div`
    position: absolute;
    bottom: -4px;
    right: -4px;
    padding: 4px;
    border-radius: 50%;
    background: #fff;
`;
const AppIcon = ({
    src,
    size,
    store,
    showStoreIcon,
}: {
    src: string,
    size: 'small' | 'medium' | 'big',
    store?: string,
    showStoreIcon?: boolean,
}) => {
    const classes = classnames({
        small: size === 'small',
        medium: size === 'medium',
        big: size === 'big',
        ios: store === 'ios',
        appletv: store === 'appletv',
    });
    if (showStoreIcon) {
        return (
            <Icon.Wrapper className={classes}>
                <Icon src={src} alt="" width="" height="" />
                <Icon.StoreOverlay>
                    <StoreIcon type={store} />
                </Icon.StoreOverlay>
            </Icon.Wrapper>
        );
    }
    return (
        <Icon.Wrapper className={classes}>
            <Icon src={src} alt="" width="" height="" />
        </Icon.Wrapper>
    );
};

/**
 * <Flag/>
 * @param {String} code
 * @example <Flag code={code} />
 */
const Flag = ({ code }: { code: string }) => {
    let size = {
        width: 16,
        height: 11,
    };
    if (code === 'ww') {
        size = {
            width: 32,
            height: 32,
        };
    }
    return (
        <Flag.Image
            src={`/static/flags/${code}.gif`}
            width={size.width}
            height={size.height}
            alt=""
        />
    );
};
Flag.Image = styled.img`
    width: 12px;
    height: auto;
`;

/**
 * <StoreIcon/>
 * @param {String} type android | ios | appletv | ibook | ga | mac  | gp  |amz | wp | win
 * @param {String} size default 12px
 * @example <StoryIcon type={type} />
 */
const StoreIcon = styled.i`
    width: ${props => (props.size ? props.size : '12px')};
    height: ${props => (props.size ? props.size : '12px')};
    display: block;
    background:${props => `url(/static/store/${props.type}.png)`} no-repeat center;
    background-position: 0 0;
    background-size: ${props => (props.size ? `${props.size} ${props.size}` : '12px 12px')};
`;

export default AppIcon;
export { Flag, StoreIcon };
