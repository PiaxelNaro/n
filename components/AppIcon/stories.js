// @flow
import React from 'react';
import { storiesOf } from '@kadira/storybook';

import Table from '../Table';
import AppIcon, { Flag, StoreIcon } from '.';
import flags from '../../data/flags';

storiesOf('AppIcon', module)
    .add('App icon', () => (
        <Table
            columns={[
                {
                    label: 'Store\\Size',
                    key: 'store',
                    isSortable: false,
                },
                {
                    label: 'Small',
                    key: 'small_icon',
                    isSortable: false,
                },
                {
                    label: 'Medium',
                    key: 'medium_icon',
                    isSortable: false,
                },
                {
                    label: 'Big',
                    key: 'big_icon',
                    isSortable: false,
                },
            ]}
            items={[
                {
                    id: 1,
                    store: 'default icon',
                    small_icon: (
                        <AppIcon
                            size="small"
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                        />
                    ),
                    medium_icon: (
                        <AppIcon
                            size="medium"
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                        />
                    ),
                    big_icon: (
                        <AppIcon
                            size="big"
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                        />
                    ),
                },
                {
                    id: 2,
                    store: 'ios icon(has border radius)',
                    small_icon: (
                        <AppIcon
                            size="small"
                            store="ios"
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                        />
                    ),
                    medium_icon: (
                        <AppIcon
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                            size="medium"
                            store="ios"
                            showStoreIcon={Boolean(true)}
                        />
                    ),
                    big_icon: (
                        <AppIcon
                            size="big"
                            store="ios"
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                        />
                    ),
                },
                {
                    id: 3,
                    store: 'apple tv icon(has border radius)',
                    small_icon: (
                        <AppIcon
                            size="small"
                            store="appletv"
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                        />
                    ),
                    medium_icon: (
                        <AppIcon
                            size="medium"
                            store="appletv"
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                            showStoreIcon={Boolean(true)}
                        />
                    ),
                    big_icon: (
                        <AppIcon
                            size="big"
                            store="appletv"
                            src="https://static-s.aa-cdn.net/img/ios/284882215/99fcb4b1f586ad70ac5306a99fc60f3a_w80"
                        />
                    ),
                },
                {
                    id: 4,
                    store: 'apple tv icon(nonsquare icon)',
                    small_icon: (
                        <AppIcon
                            size="small"
                            store="appletv"
                            src="https://static-s.aa-cdn.net/img/appletv/50600001008333/c253ddf24ba6f403906f5f816b506879_w80"
                        />
                    ),
                    medium_icon: (
                        <AppIcon
                            size="medium"
                            store="appletv"
                            showStoreIcon={Boolean(true)}
                            src="https://static-s.aa-cdn.net/img/appletv/50600001008333/c253ddf24ba6f403906f5f816b506879_w80"
                        />
                    ),
                    big_icon: (
                        <AppIcon
                            size="big"
                            store="appletv"
                            src="https://static-s.aa-cdn.net/img/appletv/50600001008333/c253ddf24ba6f403906f5f816b506879_w80"
                        />
                    ),
                },
            ]}
        />
    ))
    .add('App store icon', () => {
        const STORE_SET = {
            Android: 'android',
            'Apple TV': 'appletv',
            iBook: 'ibook',
            iOS: 'ios',
            'Google Analytic': 'google-analytic',
            Mac: 'mac',
            'Google Play': 'google-play',
            'Amazon Store': 'amazon-store',
            'Windows Phone': 'windows-phone',
            'Windows Store': 'windows-store',
        };
        const columns = [
            {
                label: 'Store Name',
                key: 'store_name',
                isSortable: false,
            },
            {
                label: 'Type',
                key: 'store_type',
                isSortable: false,
            },
            {
                label: 'Icon',
                key: 'icon',
                isSortable: false,
            },
        ];

        const injectIcon = () =>
            Object.keys(STORE_SET).map(name => ({
                id: STORE_SET[name],
                store_name: name,
                store_type: STORE_SET[name],
                icon: STORE_SET[name],
            }));

        const formatters = {
            icon: type => <StoreIcon type={type} />,
        };
        return <Table items={injectIcon()} columns={columns} formatters={formatters} />;
    })
    .add('Flag icon', () => {
        const columns = [
            {
                label: 'Country Name',
                key: 'country_name',
                isSortable: false,
            },
            {
                label: 'Country Code',
                key: 'country_code',
                isSortable: false,
            },
            {
                label: 'Flag Icon',
                key: 'flag_icon',
                isSortable: false,
            },
        ];

        const injectFlag = () => {
            const flagItems = Object.keys(flags).sort().map(name => ({
                id: name,
                country_name: name,
                country_code: flags[name],
                flag_icon: flags[name],
            }));
            flagItems.unshift({
                id: 'ww',
                country_name: 'world wide',
                country_code: 'ww',
                flag_icon: 'ww',
            });
            return flagItems;
        };
        const formatters = {
            flag_icon: code => <Flag code={code} />,
        };
        return <Table items={injectFlag()} columns={columns} formatters={formatters} />;
    });
