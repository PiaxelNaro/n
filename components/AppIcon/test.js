// @flow
import React from 'react';
import { shallow } from 'enzyme';

import AppIcon, { Flag, StoreIcon } from '.';

describe('<AppIcon>', () => {
    it('renders app icon', () => {
        const selector = shallow(
            <AppIcon src="/static/logo.png" size="small" store="ios" />
        );
        expect(selector).toMatchSnapshot();
    });
    it('renders app icon with store icon', () => {
        expect(
            shallow(
                <AppIcon
                    src="/static/logo.png"
                    size="medium"
                    store="ios"
                    showStoreIcon={Boolean(true)}
                />
            )
        ).toMatchSnapshot();
    });
});

describe('<StoreIcon>', () => {
    it('render app store icon', () => {
        expect(shallow(<StoreIcon type="android" />)).toMatchSnapshot();
        expect(shallow(<StoreIcon type="android" size="16px" />)).toMatchSnapshot();
    });
});

describe('<Flag>', () => {
    it('render flag icon', () => {
        expect(shallow(<Flag code="cn" />)).toMatchSnapshot();
    });
    it('render world wide flag', () => {
        expect(shallow(<Flag code="ww" />)).toMatchSnapshot();
    });
});
