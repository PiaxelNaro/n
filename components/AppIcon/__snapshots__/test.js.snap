// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppIcon> renders app icon 1`] = `
<AppIcon__Wrapper
  className="small ios"
>
  <AppIcon__Icon
    alt=""
    height=""
    src="/static/logo.png"
    width=""
  />
</AppIcon__Wrapper>
`;

exports[`<AppIcon> renders app icon with store icon 1`] = `
<AppIcon__Wrapper
  className="medium ios"
>
  <AppIcon__Icon
    alt=""
    height=""
    src="/static/logo.png"
    width=""
  />
  <AppIcon__StoreOverlay>
    <AppIcon__StoreIcon
      type="ios"
    />
  </AppIcon__StoreOverlay>
</AppIcon__Wrapper>
`;

exports[`<Flag> render flag icon 1`] = `
<AppIcon__Image
  alt=""
  height={11}
  src="/static/flags/cn.gif"
  width={16}
/>
`;

exports[`<Flag> render world wide flag 1`] = `
<AppIcon__Image
  alt=""
  height={32}
  src="/static/flags/ww.gif"
  width={32}
/>
`;

exports[`<StoreIcon> render app store icon 1`] = `
<i
  className="AppIcon__StoreIcon-s47nqcq-4 egjfHP"
  type="android"
/>
`;

exports[`<StoreIcon> render app store icon 2`] = `
<i
  className="AppIcon__StoreIcon-s47nqcq-4 gzjNCJ"
  size="16px"
  type="android"
/>
`;
